### PR TITLE
mgr/dashboard: disallow editing read-only config options

### PR DESCRIPTION
**Pull request by s0nea**
screenshot_2019-02-06_16-36-30

Fixes: https://tracker.ceph.com/issues/34528
Signed-off-by: Tatjana Dehler <tdehler@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by LenzGr** [comment]
Thank you for working on this. I'm not sure if it's worth devoting a column in the data table for this information. Wouldn't it be sufficient to simply disable the Edit button for these settings? The flag "Can be updated at runtime" is already displayed in the option's details when clicking on it. Maybe the cross/checkbox icons should be displayed there instead of the current "true/false" text?

**Comment by s0nea** [comment]
@LenzGr 
> Thank you for working on this. I'm not sure if it's worth devoting a column in the data table for this information. Wouldn't it be sufficient to simply disable the Edit button for these settings? The flag "Can be updated at runtime" is already displayed in the option's details when clicking on it. Maybe the cross/checkbox icons should be displayed there instead of the current "true/false" text?

Yes, in general I agree with you. I do have one concern: if the button only gets deactivated, the user may ask himself why he can't edit a specific config option. I don't think it's really clear if an option can be edited or not if the column is missing. The "Can be updated at runtime" at the bottom is too hidden from my point of view.
What do you think?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.html [hunk 1]
```diff
@@ -37,3 +37,12 @@
     </span>
   </span>
 </ng-template>
+
+<ng-template #editableTpl let-value="value">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.html]: I would suggest to use the yesNoText column that will be introduced with #25989 (see https://github.com/ceph/ceph/pull/25989/files#diff-be3e8c59695fc2d22701cff36f26dd63R8 and https://github.com/ceph/ceph/pull/25989/files#diff-246efec6cbd1bfb84c053e1067c8b65bR186).

Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.html]: I generally favor the approach where things are generalized, but this does not nearly look like the current implementation. I suggest to do that generalization and refactoring later and talk about how it should look afterwards, as I think people might want to have it look more like it currently does.

**Comment by s0nea** [comment]
screenshot_2019-02-07_15-26-49

**Comment by callithea** [comment]
jenkins test dashboard

**Comment by s0nea** [comment]
screenshot_2019-02-08_11-56-32

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts [hunk 2]
```diff
@@ -135,4 +144,12 @@ export class ConfigurationComponent implements OnInit {
     });
     this.data = [...this.data];
   }
+
+  isEditable(selection: CdTableSelection): true | false {
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]:
```suggestion
  isEditable(selection: CdTableSelection): boolean {
```

**Pull request merged**